src/utils/index.ts

```diff
@@ -0,0 +1,65 @@
+import { complement, curry, either, isEmpty, isNil, path } from 'ramda';
+import * as parse from 'url-parse';
+
+export const exists = complement(either(isEmpty, isNil));
+
+// tslint:disable-next-line:variable-name
+const _href = keys => path(['_links', ...keys, 'href']);
+
+export const href = curry((type, entity) => _href([type])(entity));
+export const self = href('self');
+
+// Each character can be neatly fitted into six bits. (<64 elements)
+//
+// We are going to place the digits and symbols haphazardly to make
+// the final output seem like an ID.
+const forward = '0123456789abcdefghijklmnopqrstuvwxyz-/_?&=';
+
+// This is the Base-64 alphabet, which is also indexed by six bits.
+// How convenient!
+const bckward = 'abcdefghijklmnopqrstuvwxyz0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+
+const forwardTable = new Map();
+
+class BadCharError extends Error {}
```

Comment:
The furl stuff seems a little specific to axiom?

Reply:
Yeah, we really shouldn't be using anything in this file.